pkg/ipfix/ipfix_intermediate.go

```diff
@@ -54,7 +58,10 @@ func (ap *ipfixAggregationProcess) Stop() {
 }
 
 func (ap *ipfixAggregationProcess) ForAllRecordsDo(callback ipfixintermediate.FlowKeyRecordMapCallBack) error {
-	return ap.AggregationProcess.ForAllRecordsDo(callback)
+	ap.mutex.Lock()
+	defer ap.mutex.Unlock()
+	err := ap.AggregationProcess.ForAllRecordsDo(callback)
+	return err
```

Comment:
I think aggregationProcess already has the mutex lock implementation. https://github.com/vmware/go-ipfix/blob/v0.4.5/pkg/intermediate/aggregate.go#L137-L139

Reply:
This is for the wrapper object created in pkg/flowaggregator/flowaggregator.go. It will be accessed by two go routines simultaneously.